x-pack/platform/plugins/shared/onechat/public/application/components/tools/execute/test_tools.tsx

```diff
@@ -0,0 +1,216 @@
+import { appPaths } from '../../../utils/app_paths';
+
+interface OnechatTestToolFlyout {
+  isOpen: boolean;
+  isLoading?: boolean;
+  toolId: string;
+  onClose: () => void;
+}
+
+interface ToolParameter {
+  name: string;
+  label: string;
+  value: string;
+  type: string;
+}
+
+const getParameters = (tool: ToolDefinition | undefined): Array<ToolParameter> => {
+  if (!tool) return [];
+
+  const fields: Array<ToolParameter> = [];
+  if (tool.configuration && tool.configuration.params) {
+    const params = tool.configuration.params as Record<string, any>;
+    Object.entries(params).forEach(([paramName, paramConfig]) => {
+      fields.push({
+        name: paramName,
+        label: paramName,
+        value: '',
+        type: paramConfig.type || 'text',
+      });
+    });
+  }
+
+  return fields;
+};
+
+export const OnechatTestFlyout: React.FC<OnechatTestToolFlyout> = ({ toolId, onClose }) => {
+  const [response, setResponse] = useState<string>('{}');
+  const { navigateToOnechatUrl } = useNavigation();
+
+  const { tool } = useTool({ toolId });
+
+  const handleClose = () => {
+    navigateToOnechatUrl(appPaths.tools.list);
+  };
+
+  const form = useForm({
+    mode: 'onChange',
+  });
+  const {
+    handleSubmit,
+    formState: { errors },
+  } = form;
+
+  const { executeTool, isLoading: isExecuting } = useExecuteTool({
+    onSuccess: (data: ExecuteToolResponse) => {
+      setResponse(JSON.stringify(data, null, 2));
+    },
+    onError: (error: Error) => {
+      setResponse(JSON.stringify({ error: error.message }, null, 2));
+    },
+  });
+
+  const onSubmit = async (formData: Record<string, any>) => {
+    const toolParams: Record<string, any> = {};
+    getParameters(tool).forEach((field) => {
+      if (field.name) {
+        let value = formData[field.name];
+        if (field.type === 'integer' || field.type === 'long') {
+          value = parseInt(value, 10);
+        } else if (field.type === 'double' || field.type === 'float') {
+          value = parseFloat(value);
+        }
+        toolParams[field.name] = value;
+      }
+    });
+
+    await executeTool({
+      toolId: tool!.id,
+      toolParams,
+    });
+  };
+
+  return (
+    <EuiFlyout onClose={handleClose} aria-labelledby="flyoutTitle">
+      <EuiFlyoutHeader hasBorder>
+        <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+          <EuiFlexItem>
+            <EuiTitle size="m">
+              <h2 id="flyoutTitle">
+                {i18n.translate('xpack.onechat.tools.testFlyout.title', {
+                  defaultMessage: 'Test Tool',
+                })}
+              </h2>
+            </EuiTitle>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <FormProvider {...form}>
+          <EuiFlexGroup gutterSize="l" responsive={false}>
+            <EuiFlexItem grow={false} style={{ minWidth: '200px', maxWidth: '300px' }}>
```

Comment:
We prefer to use `@motion` css in js I think, check the `css=` property in other files in 1chat